impeller/entity/shaders/gaussian_blur/gaussian_blur.glsl

```diff
@@ -60,14 +60,28 @@ void main() {
   f16vec4 total_color = f16vec4(0.0hf);
   float16_t gaussian_integral = 0.0hf;
 
-  for (float16_t i = -blur_info.blur_radius; i <= blur_info.blur_radius; i++) {
-    float16_t gaussian = IPGaussian(i, blur_info.blur_sigma);
-    gaussian_integral += gaussian;
-    total_color +=
-        gaussian * Sample(texture_sampler,  // sampler
-                          v_texture_coords + blur_info.blur_uv_offset *
-                                                 i  // texture coordinates
-                   );
+  for (float16_t i = 0.0hf; i <= blur_info.blur_radius; i += 2.0hf) {
+    float16_t w1 = IPGaussian(i, blur_info.blur_sigma);
+    float16_t w2 = IPGaussian(i + 1.0hf, blur_info.blur_sigma);
+    float16_t gaussian = w1 + w2;
+
+    f16vec2 offset_1 = blur_info.blur_uv_offset * i;
+    f16vec2 offset_2 = offset_1 + blur_info.blur_uv_offset;
+    vec2 pos_c1 = v_texture_coords + offset_1;
+    vec2 pos_c2 = v_texture_coords + offset_2;
+    vec2 neg_c1 = v_texture_coords - offset_1;
+    vec2 neg_c2 = v_texture_coords - offset_2;
+
+    vec2 pos_coord = (w1 * pos_c1 + w2 * pos_c2) / gaussian;
+    vec2 neg_coord = (w1 * neg_c1 + w2 * neg_c2) / gaussian;
+
+    gaussian_integral += (gaussian + gaussian);
+    total_color += gaussian * Sample(texture_sampler,  // sampler
+                                     pos_coord         // texture coordinates
+                              );
+    total_color += gaussian * Sample(texture_sampler,  // sampler
+                                     neg_coord         // texture coordinates
+                              );
```

Comment:
Leveraging linear sampling is a great idea. :)

Instead of fanning out the two sides here, I think doing something like `for (float16_t i = round(-blur_info.blur_radius); i <= blur_info.blur_radius; i += 2.hf)` would be the same?

Reply:
should we just round in the contents so blur_radius is always integral?

Reply:
Yeah rounding on the CPU seems good

Reply:
Done

Reply:
oops, where did it go...

Reply:
oh nvm, its there